### PR TITLE
Migrate java-11 sample app to use Java17

### PR DESCRIPTION
**Pull request by jwdjj**
## Description

Fixes #<ISSUE-NUMBER>

Migrate java-11 cloudstorage, websocket-jetty, datastore sample app to use Java 17 (and later). 
Java 20+ might have warning at compile but the app runs as expected

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [X] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/javniceea-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: flexible/java-17/cloudstorage/pom.xml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  <artifactId>flexible-cloudstorage</artifactId>
+
+  <!--
+    The parent pom defines common style checks and testing strategies for our samples.
+    Removing or replacing it should not affect the execution of the samples in anyway.
+  -->
+  <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.2.0</version>
```
Comment (iennae) [inline flexible/java-17/cloudstorage/pom.xml:30]: 1.2.2 to pull latest. https://central.sonatype.com/artifact/com.google.cloud.samples/shared-configuration

Comment (jwdjj) [reply flexible/java-17/cloudstorage/pom.xml]: This is updated now, the checks will run

Everything still works correctly, using the latest dependency does resolve the jacoco dependency warning during compilation as well (tested with Java 17 and Java 22). Thanks.

**Review by ludoch — APPROVED** [review]

File: flexible/java-17/cloudstorage/pom.xml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.2.2</version>
+  </parent>
+
+  <properties>
+    <maven.compiler.target>17</maven.compiler.target>
+    <maven.compiler.source>17</maven.compiler.source>
+    <failOnMissingWebXml>false</failOnMissingWebXml> <!-- REQUIRED -->
+    <appengine.maven.plugin>2.8.0</appengine.maven.plugin>
```
Comment (ludoch) [inline flexible/java-17/cloudstorage/pom.xml:37]: 2.8.1 now?

File: flexible/java-17/datastore/pom.xml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    <version>1.2.2</version>
+  </parent>
+
+  <properties>
+    <maven.compiler.target>17</maven.compiler.target>
+    <maven.compiler.source>17</maven.compiler.source>
+
+    <failOnMissingWebXml>false</failOnMissingWebXml> <!-- REQUIRED -->
+
+    <appengine.maven.plugin>2.8.0</appengine.maven.plugin>
```
Comment (ludoch) [inline flexible/java-17/datastore/pom.xml:39]: 2.8.1

**Review by ludoch — APPROVED** [review]
Do we have public doc links to these samples?
Did you deploy to make sure all is good?

**Comment by jwdjj** [comment]
> Do we have public doc links to these samples? Did you deploy to make sure all is good?

Yes and I have test deployment with the updated dependency as well

[Cloud storage](https://cloud.google.com/appengine/docs/flexible/using-cloud-storage.md?tab=java#top)
[Datastore](https://cloud.google.com/appengine/docs/flexible/using-firestore-in-datastore-mode.md?tab=java#top)
[Websocket](https://cloud.google.com/appengine/docs/flexible/using-websockets-and-session-affinity?tab=java)

**Review by ludoch — APPROVED** [review]

**Pull request merged**